### PR TITLE
feat: write to ecs7 components

### PR DESCRIPTION
**Pull request by pbosio**
add code to be able to create/modify/remove component's data.
modification will be applied locally and sent to kernel as a binary message.

### CODE REVIEW

**Review by Suduck — APPROVED** [review]

**Review by Suduck — APPROVED** [review]

File: unity-renderer/Assets/DCLPlugins/ECS7/ComponentCrdtWriteSystem/ComponentCrdtWriteSystem.cs [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    {
+        if (queuedMessages.Count == 0)
+        {
+            return;
+        }
+
+        Queue<CRDTMessage> sceneMessages;
+
+        // we prioritize current scene's messages
+        string currentSceneId = CommonScriptableObjects.sceneID.Get();
```
Comment (AjimenezDCL) [inline unity-renderer/Assets/DCLPlugins/ECS7/ComponentCrdtWriteSystem/ComponentCrdtWriteSystem.cs:71]: These CommonScriptableObjects are as slow as ugly, we must port the `currentSceneId` to a DataStore soon. 

(No a request, just a heads up for the future).

File: unity-renderer/Assets/DCLPlugins/ECS7/ComponentCrdtWriteSystem/ComponentCrdtWriteSystem.cs [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+        if (!queuedMessages.TryGetValue(sceneId, out Queue<CRDTMessage> sceneMessages))
+        {
+            sceneMessages = new Queue<CRDTMessage>();
+            queuedMessages.Add(sceneId, sceneMessages);
+        }
+
+        sceneMessages.Enqueue(message);
+    }
+
+    internal void ProcessMessages()
```
Comment (AjimenezDCL) [inline unity-renderer/Assets/DCLPlugins/ECS7/ComponentCrdtWriteSystem/ComponentCrdtWriteSystem.cs:61]: Should this be an async function to be throttled? maybe we can have a IMessageProcessor so we dont have to decide now

File: unity-renderer/Assets/DCLPlugins/ECS7/ComponentCrdtWriteSystem/ComponentCrdtWriteSystem.cs [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+        }
+
+        while (messages.Count > 0)
+        {
+            KernelBinaryMessageSerializer.Serialize(binaryWriter, messages.Dequeue());
+
+            if (memoryStream.Length >= BINARY_MSG_MAX_SIZE && messages.Count > 0)
+            {
+                DispatchBinaryMessage(sceneId, memoryStream);
+                return false;
```
Comment (AjimenezDCL) [inline unity-renderer/Assets/DCLPlugins/ECS7/ComponentCrdtWriteSystem/ComponentCrdtWriteSystem.cs:120]: Isn't this just dispatching a message per frame? isn't it slow?

File: unity-renderer/Assets/DCLPlugins/ECS7/ECS7Plugin.cs [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    private readonly ComponentCrdtWriteSystem crdtWriteSystem;
+    private readonly IECSComponentWriter componentWriter;
+    private readonly ECS7ComponentsPlugin componentsPlugin;
+
+    public ECS7Plugin()
+    {
+        crdtWriteSystem = new ComponentCrdtWriteSystem(Environment.i.platform.updateEventHandler, Environment.i.world.state);
+        componentWriter = new ECSComponentWriter(crdtWriteSystem.WriteMessage);
+
+        componentsPlugin = new ECS7ComponentsPlugin(DataStore.i.ecs7.componentsFactory, componentWriter);
```
Comment (AjimenezDCL) [inline unity-renderer/Assets/DCLPlugins/ECS7/ECS7Plugin.cs:15]: The `componentFactory` isn't needed by anybody else (unless I'm mistaken). I would just pass it through a constructor here or even better just make ECS7Plugin to instantiate it (if possible).

**Review by AjimenezDCL — APPROVED** [review]
Looks good, added some minor requestes

File: unity-renderer/Assets/DCLPlugins/ECS7/ECS7Plugin.cs (hunk 4 above)
Comment (pbosio) [reply unity-renderer/Assets/DCLPlugins/ECS7/ECS7Plugin.cs:15]: the factory is used by another system that is not inside this plugin.
I'm waiting for be able to do a benchmark to see if I can move that system inside this plugin

File: unity-renderer/Assets/DCLPlugins/ECS7/ComponentCrdtWriteSystem/ComponentCrdtWriteSystem.cs (hunk 2 above)
Comment (AjimenezDCL) [reply unity-renderer/Assets/DCLPlugins/ECS7/ComponentCrdtWriteSystem/ComponentCrdtWriteSystem.cs:61]: Fair enough!

File: unity-renderer/Assets/DCLPlugins/ECS7/ECS7Plugin.cs (hunk 4 above)
Comment (AjimenezDCL) [reply unity-renderer/Assets/DCLPlugins/ECS7/ECS7Plugin.cs:15]: I would still avoid using `DataStore.i` and use Dependency Injection, it helps with the tests. I leave it up to you!

**Pull request merged**